include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -231,6 +230,70 @@ class is_operational_impl
         // if we made it here, the layout is operational
         return operational_status::OPERATIONAL;
     }
+    /**
+     * Determines if the given charge distribution fulfills the correct logic based on the provided charge index and
+     * truth table.
+     *
+     * @param given_cds The charge distribution surface to be checked.
+     * @param charge_index Charge index represented by the position of perturbers.
+     * @return Operational status indicating if the layout is `operational` or `non-operational`.
+     */
+    [[nodiscard]] operational_status
+    is_given_cds_operational_for_pattern(const charge_distribution_surface<Lyt>& given_cds,
+                                         const uint64_t                          charge_index) noexcept
```

Comment:
What is called `charge_index` here was called `input_pattern` in `does_charge_distribution_match_logic_for_input_pattern`. This can lead to confusion if these are even the same thing or if there is a bug in the function call (passing the wrong argument).